core/base/src/TDirectory.cxx

```diff
@@ -1028,9 +1028,12 @@ void TDirectory::FillFullPath(TString& buf) const
 ///    gDirectory->cd("b");
 ///    gDirectory->mkdir("d");
 /// ~~~
+/// returnExistingDirectory is just for compatibility with the TDirectoryFile override
 
-TDirectory *TDirectory::mkdir(const char *name, const char *title)
+TDirectory *TDirectory::mkdir(const char *name, const char *title, Bool_t returnExistingDirectory)
 {
+   if (returnExistingDirectory)
```

Comment:
Implementing (rather than reject) the desired behavior might as simple as
```
 if (returnExistingDirectory) {
      auto existingdir = GetDirectory(name);
      if (existingdir)
          return existingdir;
}
```

Reply:
Good idea. I've implemented it, although it doesn't actually change the behavior described [here](https://github.com/root-project/root/pull/4649#discussion_r350876264). (Unclear to me how TDirectory is supposed to behave when it's not a TDirectoryFile...)

Reply:
>  although it doesn't actually change the behavior described here. 

It shouldn't :) but it should also have this (new) behavior:
```
In [3]: sd = d.mkdir('daughterdir')

In [4]: sd2 = d.mkdir('daughterdir', True)

In [5]: sd
Out[5]: <ROOT.TDirectory object ("daughterdir") at 0x7fa487cf9c90>

In [6]: sd2
Out[6]: <ROOT.TDirectory object ("daughterdir") at 0x7fa487cf9c90>
```

Reply:
I'm sorry--of course, you're right. What I get for working on too many things at once:

```python
In [1]: import ROOT

In [2]: d = ROOT.TDirectory('motherdir', 'motherdir')

In [3]: sd = d.mkdir('daughterdir')

In [4]: sd2 = d.mkdir('daughterdir', '', True)

In [5]: sd
Out[5]: <ROOT.TDirectory object ("daughterdir") at 0x7fd7e0c19da0>

In [6]: sd2
Out[6]: <ROOT.TDirectory object ("daughterdir") at 0x7fd7e0c19da0>

In [7]: sd is sd2
Out[7]: True
```